File: controlweave/mobile/android/ControlWeave/app/src/main/kotlin/com/controlweave/app/core/models/DashboardModels.kt
package com.controlweave.app.core.models

data class DashboardStats(
    val totalControls: Int,
    val implementedControls: Int,
    val complianceScore: Double,
    val frameworkBreakdown: List<FrameworkBreakdown>?,
    val openFindings: Int?,
    val overdueItems: Int?
)

data class FrameworkBreakdown(
    val id: String,
    val name: String,
    val shortName: String?,
    val complianceScore: Double,
    val totalControls: Int,
    val implementedControls: Int
)

File: controlweave/mobile/ios/ControlWeave/ControlWeave/Core/Models/DashboardModels.swift
import Foundation

struct DashboardStatsResponse: Decodable {
    let success: Bool
    let data: DashboardStats
}

struct DashboardStats: Decodable {
    let totalControls: Int
    let implementedControls: Int
    let complianceScore: Double
    let frameworkBreakdown: [FrameworkBreakdown]?
    let openFindings: Int?
    let overdueItems: Int?
}

struct FrameworkBreakdown: Decodable, Identifiable {
    let id: String
    let name: String
    let shortName: String?
    let complianceScore: Double
    let totalControls: Int
    let implementedControls: Int
}
